specifyweb/specify/api.py

```diff
@@ -252,6 +277,36 @@ def collection_dispatch(request, model) -> HttpResponse:
 
     return resp
 
+def collection_dispatch_bulk(request, model) -> HttpResponse:
```

Comment:
Something to consider: lock_tables will commit the current transaction. we'll not have inconsistent states but it is not ideal to keep a connection and committing multiple times (from the user's perspective, the records should be atomic...). But, this was also a problem before so no code changes are necessary